services/metadata_service/api/tag.py

```diff
@@ -0,0 +1,139 @@
+from services.data import TaskRow
+from services.data.db_utils import DBResponse
+from services.data.postgres_async_db import AsyncPostgresDB
+from services.metadata_service.api.utils import format_response, \
+    handle_exceptions
+import json
+
+import asyncio
+
+
+class TagApi(object):
+    lock = asyncio.Lock()
+
+    def __init__(self, app):
+        app.router.add_route(
+            "POST",
+            "/tags",
+            self.update_tags,
+        )
```

Comment:
Is there a future-proofing reason why we want a single route that can update multiple resource tags in bulk, instead of the more RESTful approach of accepting a whitelisted PUT/PATCH to the specific resource routes? f.ex. updating a tasks tags could be done with a 
```PATCH /flows/{flow_id}/runs/{run_number}/steps/{step_name}/tasks/{task_id}```

Or if it is necessary to have a specific tag update route (even though tags are part of the resource in question and not a separate entry)
```PATCH /flows/{flow_id}/runs/{run_number}/steps/{step_name}/tasks/{task_id}/tags```

Reply:
On further thought this approach would lead to an immense number of requests in order to update tags for all tasks of a run. Feel free to ignore :)

Reply:
Ya, I think the idea was to allow bulk updates and since the tags can be added at lots of level it may make a lot of endpoints/calls for essentially the same thing.